### PR TITLE
KAFKA-10878 Check failed message in ProtocolSerializationTest

### PR DESCRIPTION
**Pull request by g1geordie**
use assert to check failed message 

-  testReadWhenOptionalDataMissingAtTheEndIsNotTolerated
-  testReadWithMissingNonOptionalExtraDataAtTheEnd

### CODE REVIEW

**Comment by g1geordie** [comment]
@chia7712  can you help my take a look?

**Review by chia7712 — APPROVED** [review]
@g1geordie Thanks for your patch. LGTM

File: clients/src/test/java/org/apache/kafka/common/protocol/types/ProtocolSerializationTest.java [hunk 1]
```diff
@@ -414,7 +416,7 @@ public void testReadWhenOptionalDataMissingAtTheEndIsNotTolerated() {
         oldFormat.writeTo(buffer);
         buffer.flip();
         SchemaException e = assertThrows(SchemaException.class, () -> newSchema.read(buffer));
-        e.getMessage().contains("Error reading field 'field2': java.nio.BufferUnderflowException");
+        assertThat(e.getMessage(), containsString("Error reading field 'field2': java.nio.BufferUnderflowException"));
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/protocol/types/ProtocolSerializationTest.java]: Is this error message from java standard library? If so, does it got different on different location (i.e different language)?

Comment (g1geordie) [reply clients/src/test/java/org/apache/kafka/common/protocol/types/ProtocolSerializationTest.java]: The SchemaException is throw by KAFKA. 
it's message is  `fix string + field info   +  e.msg || e.className` 
the lastest string here is className   

but It's seems to me that only check the prefix is more better

Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/common/protocol/types/ProtocolSerializationTest.java]: You are right. Checking ```Error reading field 'field2'``` should be enough for this test case.

**Pull request merged**